homeassistant/components/climate/dyson.py

```diff
@@ -0,0 +1,222 @@
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def name(self):
+        """Return the display name of this climate."""
+        return self._device.name
+
+    @property
+    def entity_id(self):
+        """Return the entity id of this climate."""
+        return "climate.{}".format(self._device.name)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self.temp_unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._device.environmental_state:
+            temperature_kelvin = self._device.environmental_state.temperature
+            if temperature_kelvin != 0:
+                if self.temp_unit == TEMP_CELSIUS:
+                    self._current_temp = float("{0:.1f}".format(
+                        self.kelvin_to_celsius(temperature_kelvin)))
+                else:
+                    self._current_temp = float("{0:.1f}".format(
+                        self.kelvin_to_fahrenheit(temperature_kelvin)))
+        return self._current_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        if self._target_temp is None:
+            heat_target = int(self._device.state.heat_target) / 10
+            if self.temp_unit == TEMP_CELSIUS:
+                return int(self.kelvin_to_celsius(heat_target))
+            return int(self.kelvin_to_fahrenheit(heat_target))
+        return self._target_temp
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        if self._device.environmental_state:
+            if self._device.environmental_state.humidity == 0:
+                return None
+            return self._device.environmental_state.humidity
+        return None
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        from libpurecoollink.const import HeatMode, HeatState
+        if self._device.state.heat_mode == HeatMode.HEAT_ON:
+            if self._device.state.heat_state == HeatState.HEAT_STATE_ON:
+                return STATE_HEAT
+            return STATE_IDLE
+        return STATE_COOL
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [STATE_HEAT, STATE_COOL]
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        from libpurecoollink.const import FocusMode
+        if self._device.state.focus_mode == FocusMode.FOCUS_ON:
+            return STATE_FOCUS
+        return STATE_DIFFUSE
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return [STATE_FOCUS, STATE_DIFFUSE]
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if kwargs.get(ATTR_TEMPERATURE) is None:
```

Comment:
Save the temperature in the first lookup to avoid having to do the lookup twice. Then check if the temperature is None.